core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -203,17 +209,29 @@ protected void add(DataFile file) {
    * Add all files in a manifest to the new snapshot.
    */
   protected void add(ManifestFile manifest) {
-    // the manifest must be rewritten with this update's snapshot ID
-    try (ManifestReader reader = ManifestReader.read(
-        ops.io().newInputFile(manifest.path()), ops.current().specsById())) {
-      ManifestFile manifestFile = ManifestWriter.copyAppendManifest(
-          reader, manifestPath(manifestCount.getAndIncrement()), snapshotId(), appendedManifestsSummary);
-      appendManifests.add(manifestFile);
-      // keep reference of the first appended manifest, so that we can avoid merging first bin(s)
-      // which has the first appended manifest and have not crossed the limit of minManifestsCountToMerge
-      if (firstAppendedManifest == null) {
-        firstAppendedManifest = manifestFile;
-      }
+    ManifestFile appendedManifest;
+    if (snapshotIdInheritanceEnabled && manifest.snapshotId() == null) {
```

Comment:
Looks like snapshots that were written with a version that used -1 will still rewrite those snapshots so we preserve the existing behavior. +1.